torch/_dynamo/convert_frame.py

```diff
@@ -375,6 +375,10 @@ def format_guard_failures(code):
 
         compile_id = CompileId(frame_id, frame_compile_id)
 
+        # Always capture dynamic shape ops in the first attempt.
+        # Switch to graph break if the first attempt failed.
+        frame_state["capture_dynamic_output_shape_ops"] = True
```

Comment:
We probably need some dedicated structure shared across restarts. This isn't the right spot because it is persisted across all compilations of the frame, which is too long an extent.

Reply:
What do you think about a global state? But we still need to set it to True at the beginning of every frame evaluation.

Reply:
Let's put something on torch/_guards.py. Maybe CompileContext which IS shared across restarts.